File: go/booking-up-for-beauty/booking_up_for_beauty.go
package booking

import "time"

// Schedule returns a time.Time from a string containing a date.
func Schedule(date string) time.Time {
	layout := "1/02/2006 15:04:05"
	appointment, err := time.Parse(layout, date)
	if err != nil {
		panic(err)
	}
	return appointment
}

// HasPassed returns whether a date has passed.
func HasPassed(date string) bool {
	appointment, err := time.Parse("January 2, 2006 15:04:05", date)
	if err != nil {
		panic(err)
	}
	return appointment.Before(time.Now())
}

// IsAfternoonAppointment returns whether a time is in the afternoon.
func IsAfternoonAppointment(date string) bool {
	appointment, err := time.Parse("Monday, January 2, 2006 15:04:05", date)
	if err != nil {
		panic(err)
	}
	if appointment.Hour() >= 12 && appointment.Hour() < 18 {
		return true
	} else {
		return false
	}
}

// Description returns a formatted string of the appointment time.
func Description(date string) string {
	appointment, err := time.Parse("1/2/2006 15:04:05", date)
	if err != nil {
		panic(err)
	}
	return appointment.Format("You have an appointment on Monday, January 2, 2006, at 15:04.")
}

// AnniversaryDate returns a Time with this year's anniversary.
func AnniversaryDate() time.Time {
	return time.Date(time.Now().Year(), time.September, 15, 0, 0, 0, 0, time.UTC)
}

File: fsharp/booking-up-for-beauty/BookingUpForBeauty.fs
module BookingUpForBeauty

// The following line is needed to use the DateTime type
open System


let schedule (appointmentDateDescription: string): DateTime =
    DateTime.Parse appointmentDateDescription


let hasPassed (appointmentDate: DateTime): bool =
    DateTime.Now > appointmentDate


/// Checks appointment is between Noon to 6.
let isAfternoonAppointment (appointmentDate: DateTime): bool =
    appointmentDate.Hour >= 12 && appointmentDate.Hour < 18


/// takes an appointment date and returns a description of that date and time:
/// "You have an appointment on 3/29/2019 3:00:00 PM."
let description (appointmentDate: DateTime): string =
    sprintf "You have an appointment on %O." appointmentDate


/// returns this year's anniversary date, which is September 15th
let anniversaryDate(): DateTime = DateTime(DateTime.Now.Year, 9, 15)